File: frontend/src/app/model/car.ts
export interface Car {
    id: string;
    customerId: string;
    make: string;
    model: string;
    year: string;
    license: string;
    date: Date;
    photoId: string;
}

export interface CarDTO {
    customerId: string;
    make: string;
    model: string;
    year: string;
    license: string;
    date: Date;
    photoId: string;
}

export interface Photo {
    id: string;
    name: string;
    type: string;
    data: any;
}

File: backend/src/main/java/krisschaaf/schuettieshome/car/model/Car.java
package krisschaaf.schuettieshome.car.model;

import krisschaaf.schuettieshome.customer.Customer;
import lombok.Data;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

import java.util.Date;

@Document
@Data
public class Car {
    @Id
    String id;

    String make;
    String model;
    String year;
    String license;
    Date date;

    @DBRef
    Customer customer;
    @DBRef
    Photo photo;

}
